geo-types/src/rect.rs

```diff
@@ -170,7 +170,7 @@ impl<T: CoordNum> Rect<T> {
     /// assert_eq!(rect.width(), 10.);
     /// ```
     pub fn width(self) -> T {
-        self.max().x - self.min().x
+        self.clone().max().x - self.min().x
```

Comment:
```suggestion
        self.max().x.clone() - self.min().x
```

(and similarly throughout this file)